tools/fuzzer/lldb_eval_libfuzzer_test.cc

```diff
@@ -0,0 +1,163 @@
+}
+
+// Generation tree visitor used to randomly pick nodes.
+class GenNodePicker : public GenTreeVisitor {
+ public:
+  void visit_node(std::shared_ptr<GenNode> node) {
+    if (node->is_valid()) {
+      options_.emplace_back(node);
+    }
+  }
+
+  template <class Rng>
+  std::shared_ptr<GenNode> pick(Rng& rng) {
+    std::uniform_int_distribution<size_t> distr(0, options_.size() - 1);
+    return options_[distr(rng)];
+  }
+
+ private:
+  std::vector<std::shared_ptr<GenNode>> options_;
+};
+
+template <class Rng>
+std::shared_ptr<GenNode> pick_random_node(std::shared_ptr<GenNode> root,
+                                          Rng& rng) {
+  GenNodePicker picker;
+  walk_gen_tree(root, &picker);
+  return picker.pick(rng);
+}
+
+// Generation tree visitor used to convert sequence of `rand_t` to byte
+// sequence representing a libFuzzer input.
+class GenNodeWriter : public GenTreeVisitor {
+ public:
+  explicit GenNodeWriter(LibfuzzerWriter& writer) : writer_(writer) {}
+
+  void visit_random_value(rand_t value) { writer_.write(value); }
+
+ private:
+  LibfuzzerWriter& writer_;
+};
+
+void write_node(std::shared_ptr<GenNode> root, LibfuzzerWriter& writer) {
+  GenNodeWriter node_writer(writer);
+  walk_gen_tree(root, &node_writer);
+}
+
+extern "C" size_t LLVMFuzzerCustomMutator(uint8_t* data, size_t size,
+                                          size_t max_size, unsigned int seed) {
+  LibfuzzerReader reader(data, size);
+  auto fixed_generator = create_generator(LibfuzzerRng(std::move(reader)));
+  auto maybe_expr = fixed_generator.generate();
+  assert(maybe_expr.has_value() && "Expression couldn't be generated!");
+
+  std::mt19937 rng(seed);
+  auto root = fixed_generator.node();
+  auto mutable_node = pick_random_node(root, rng);
+
+  auto random_generator = create_generator(Mt19937(rng()));
+  if (!random_generator.mutate_gen_node(mutable_node)) {
+    return size;
+  }
+
+  LibfuzzerWriter writer(data, max_size);
+  write_node(root, writer);
+
+  // It's possible that `root`'s sequence of random values overflows the size of
+  // `data`. Overflowed values will be ignored. This isn't ideal, but also isn't
+  // critical since expression generator is able to generate an expression from
+  // any byte sequence.
+  // TODO: Compare number of random values in the `root` to the `max_size`.
+
+  return writer.size();
+}
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  auto generator = create_generator(LibfuzzerRng(LibfuzzerReader(data, size)));
+  auto maybe_expr = generator.generate();
+  assert(maybe_expr.has_value() && "Expression couldn't be generated!");
+
+  std::ostringstream os;
+  os << maybe_expr.value();
+
+  lldb::SBError error;
+  lldb_eval::EvaluateExpression(g_frame, os.str().c_str(), error);
+  return 0;
+}
+
+extern "C" int LLVMFuzzerInitialize(int* argc, char*** argv) {
+  assert(*argc >= 1);
+  g_runfiles = Runfiles::Create((*argv)[0]);
```

Comment:
Runfiles is used only during the initialization, so it doesn't have to be a global variable. This also avoid a memory leak, since currently `g_runtimes` is never deleted .

Reply:
Done, also turned `g_process` to local variable.